### PR TITLE
Start building instrumentation for scrapers

### PR DESCRIPTION
**Pull request by johnseekins**
We'll use statsd-style metrics (aggregation managed on the other end) to let us emit data as scrapers are running (potentially). 
Using https://github.com/civic-eagle/statsd-http-proxy as the receiver lets us actually require auth on sending.

Example stats produced from these changes for a bill scrape in WY:

```
[{'value': 2, 'metric': 'openstates_active_sessions', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming'},
{'value': 0, 'metric': 'openstates_unaccounted_sessions', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming'},
{'value': 11, 'metric': 'openstates_ignored_sessions', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming'},
{'value': 1, 'metric': 'openstates_jurisdiction_scrapes_total', 'metric_type': 'count', 'tags': 'jurisdiction=Wyoming'},
{'value': 1, 'metric': 'openstates_session_scrapes_total', 'metric_type': 'count', 'tags': 'jurisdiction=Wyoming,session=2022'},
{'value': 1669928209.5008867, 'metric': 'openstates_last_session_scrape_time', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,session=2022'},
{'value': 1, 'metric': 'openstates_session_scrapes_total', 'metric_type': 'count', 'tags': 'jurisdiction=Wyoming,session=2021S1'},
{'value': 1669928252.2023642, 'metric': 'openstates_last_session_scrape_time', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,session=2021S1'},
{'value': 1669928259.956057, 'metric': 'openstates_last_collection_run_time', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming'},
{'value': 342, 'metric': 'openstates_scrape_runtime', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=jurisdiction'},
{'value': 1, 'metric': 'openstates_objects_collected', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=jurisdiction,object_type=jurisdiction'},
{'value': 3, 'metric': 'openstates_objects_collected', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,object_type=organization,scrape_type=jurisdiction'},
{'value': 342, 'metric': 'openstates_scrape_runtime_secs', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=bills'},
{'value': 1089, 'metric': 'openstates_objects_collected', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,object_type=vote_event,scrape_type=bills'},
{'value': 320, 'metric': 'openstates_objects_collected', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=bills,object_type=bill'},
{'value': 0, 'metric': 'openstates_objects_imported', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,import_type=insert,scrape_type=jurisdiction'},
{'value': 0, 'metric': 'openstates_objects_imported', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=jurisdiction,import_type=update'},
{'value': 1, 'metric': 'openstates_objects_imported', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=jurisdiction,import_type=noop'},
{'value': 0, 'metric': 'openstates_objects_imported', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=bill,import_type=insert'},
{'value': 0, 'metric': 'openstates_objects_imported', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=bill,import_type=update'},
{'value': 320, 'metric': 'openstates_objects_imported', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=bill,import_type=noop'},
{'value': 0, 'metric': 'openstates_objects_imported', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,import_type=insert,scrape_type=vote_event'},
{'value': 0, 'metric': 'openstates_objects_imported', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=vote_event,import_type=update'},
{'value': 1089, 'metric': 'openstates_objects_imported', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=vote_event,import_type=noop'},
{'value': 0, 'metric': 'openstates_objects_imported', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=event,import_type=insert'},
{'value': 0, 'metric': 'openstates_objects_imported', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=event,import_type=update'},
{'value': 0, 'metric': 'openstates_objects_imported', 'metric_type': 'gauge', 'tags': 'jurisdiction=Wyoming,scrape_type=event,import_type=noop'}]
```

### CODE REVIEW

File: openstates/utils/instrument.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        return jwt.encode({"id": "openstates"}, secret, algorithm="HS256")
+
+    def _send_stats(self, force: bool = False) -> None:
+        """
+        Needs to be broken out from _process_metric to have a
+        "write at shutdown" function (force=True)
+        """
+        batch_len = len(self._batch)
+        if (force and batch_len > 0) or batch_len > self.batch_size:
+            if not self.endpoint:
```
Comment (jessemortenson) [inline openstates/utils/instrument.py:76]: Am I understanding this correctly: if a 3rd party uses the openstates-core CLI, and doesn't have stats configured, the CLI commands will work normally. Except they will also emit a warning for every attempt to send stats (which could be what, like a handful of them? or dozens of them?). Is that right?

If so, seems like that meets a criteria I have (folks can still use the openstates CLI without having to have the full Civic Eagle infra in place. 

It does seem like maybe one "stats are not enaled) warning at the start or end of command execution might be nicer than many of them.

**Review by jessemortenson — COMMENTED** [review]
I don't think I'm the right person to offer a approve/deny on this as I'm not terribly close to the openstates-core code. Happy to provide feedback at a high level though. Do you have a list of which stats you want to emit for scrapers somewhere I could look at? That would be easier for me than parsing that out of the code

File: openstates/utils/instrument.py (hunk 1 above)
Comment (johnseekins) [reply openstates/utils/instrument.py]: Good point. Changed the logging to only output at debug level when configs aren't set. Should make normal runs much cleaner for people.

**Review by NewAgeAirbender — APPROVED** [review]

**Review by bfossen-ce — APPROVED** [review]
These changes LGTM. The changes will be really beneficial and instrument.py was well written and easy to follow

**Pull request merged**